hibernate6-jpa/src/main/java/io/micronaut/configuration/hibernate6/jpa/proxy/IntrospectedHibernateBytecodeProvider.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.hibernate6.jpa.proxy;
+
+import io.micronaut.core.annotation.Internal;
+import jakarta.inject.Singleton;
+import org.hibernate.bytecode.enhance.spi.EnhancementContext;
+import org.hibernate.bytecode.enhance.spi.Enhancer;
+import org.hibernate.bytecode.spi.BytecodeProvider;
+import org.hibernate.bytecode.spi.ProxyFactoryFactory;
+import org.hibernate.bytecode.spi.ReflectionOptimizer;
+import org.hibernate.property.access.spi.PropertyAccess;
+
+import java.util.Map;
+
+/**
+ * Compile-time proxies implementation of Hibernate's {@link BytecodeProvider}.
+ * Implementation provides custom implementation of {@link ProxyFactoryFactory} and disables bytecode enhancer.
+ *
+ * @author Denis Stepanov
+ * @since 3.3.0
+ */
+@Singleton
+@Internal
+public final class IntrospectedHibernateBytecodeProvider implements BytecodeProvider {
+
+    private static final Enhancer NO_OP = (className, originalBytes) -> null;
+
+    @Override
+    public ProxyFactoryFactory getProxyFactoryFactory() {
+        return new IntroducedHibernateProxyFactoryFactory();
+    }
+
+    @Override
+    public ReflectionOptimizer getReflectionOptimizer(Class clazz, String[] getterNames, String[] setterNames, Class[] types) {
+        // This is deprecated and no longer used
+        return null;
+    }
+
+    @Override
+    public ReflectionOptimizer getReflectionOptimizer(Class<?> clazz, Map<String, PropertyAccess> propertyAccessMap) {
+        // Prev implementation doesn't return accurate optimizer and bean properties
+        // So some tests are failing. Returning null fixes some failing tests for now
+        return null;
```

Comment:
It appears this never worked in prev micronaut-sql versions as method was this
```
@Override
                    public Object[] getPropertyValues(Object object) {
                        Object[] values = new Object[beanProperties.length];
                        for (int i = 0; i < beanProperties.length; i++) {
                            BeanProperty beanProperty = beanProperties[i];
                            values[i] = beanProperty.get(i);
                        }
                        return values;
                    }
```
This line `values[i] = beanProperty.get(i);` should be `values[i] = beanProperty.get(object );` or otherwise it would fail. But, when I change it then still getting errors since this doesn't return matching field names with what Hibernate expects. This for instance returns 3 fields in one of our tests, hibernate expects 2 and in different order. Since it didn't work before, I guess we can ignore it. The reason it was not worked is flag `hibernate.bytecode.use_reflection_optimizer` in hibernate that was turned off by default in hibernate5 and now is enabled in hibernate6. And we would get issue reported if someone had it enabled.